Adding setters for system properties

jbescos opened the pull request:
Relates to https://github.com/eclipse-ee4j/mail/issues/513

Adding those parameters in constructor bring some problems:

- These classes already have some constructors and we cannot modify them. So I would need to create duplicates to add these parameters.
- Doesn't look very well when there are many parameters to be set.

So I thought about keeping constructors and creating a builder. Then setting parameters is easy, but it has the disadvantage that scales badly with heritage. These classes are somehow thought to be extended.

And finally, the thing that fits better in this scenario is adding some setters. It has the disadvantage that it is not thread safe, but anyway there are other fields that are not thread safe already. The user must know that same instances of this class are not thought to be modified by different threads.

chrisrueger commented:
Looks like a reasonable fix for #513 👍

chrisrueger commented:
maybe you can join the discussion in #513 ? there is an alternativ proposal. @jbescos

jbescos commented:
@chrisrueger I read it and I will need to try it. I am busy with other tasks right now but I will come back to this. I am converting this PR to draft for the time being.

jbescos commented:
I have added changes related to session properties

jbescos commented:
There are 2 tests failing:
```
FAILED........javasoft/sqe/tests/jakarta/mail/Message/setFileNameTest_encodeFalse_decodeTrue.java#setFileNameTest_encodeFalse_decodeTrue
FAILED........com/sun/tdk/signaturetest/SignatureTest.java
```

But I don't see where are them. I need to run them to know what is the problem, does somebody know it?.

lukasj commented:
In the mail-tck repo

jbescos commented:
There is a signature test failing. It complains with the added fields and the constructor. Do we modify that test to be fine with that?.

There is other test failing that I need to investigate: api/javasoft/sqe/tests/jakarta/mail/Message/testlist.html#setFileNameTest_encodeFalse_decodeTrue: setFileName() API(s) with mail.mime.encodefilename=false and mail.mime.decodefilename=true 

```
Added Fields
------------

jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.allowutf8
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.cacheMultipart
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.decodeFileName
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.encodeFileName
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.ignoreMultipartEncoding
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.setContentTypeFileName
jakarta.mail.internet.MimeBodyPart:     field protected boolean jakarta.mail.internet.MimeBodyPart.setDefaultTextCharset
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.allowutf8
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.cacheMultipart
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.decodeFileName
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.encodeFileName
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.ignoreMultipartEncoding
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.setContentTypeFileName
jakarta.mail.internet.MimeMessage:      field protected boolean jakarta.mail.internet.MimeMessage.setDefaultTextCharset
jakarta.mail.internet.MimePartDataSource:                   field protected boolean jakarta.mail.internet.MimePartDataSource.ignoreMultipartEncoding

Added Constructors
------------------

jakarta.mail.internet.MimeBodyPart:     constructor public jakarta.mail.internet.MimeBodyPart.<init>(jakarta.mail.Session,java.io.InputStream) throws jakarta.mail.MessagingException

duplicate messages suppressed: 7

STATUS:Failed.16 errors
----------out2:(0/0)----------
result: Failed. 16 errors


test result: Failed. 16 errors
```

jbescos commented:
I investigated this test:
api/javasoft/sqe/tests/jakarta/mail/Message/testlist.html#setFileNameTest_encodeFalse_decodeTrue

The issue is that the system properties are set after MimeMessage is instanced, so that properties are ignored because MimeMessage makes use of them.

I can make some changes in mail to make it work, so we don't need to modify the TCK test.

jbescos commented:
I have created a https://github.com/eclipse-ee4j/mail-tck/pull/24 for the issue of: api/javasoft/sqe/tests/jakarta/mail/Message/testlist.html#setFileNameTest_encodeFalse_decodeTrue

Now it is failing only the SignatureTest, but I need advise about how to proceed.
